.github/workflows/build.yml

```diff
@@ -1,28 +1,32 @@
 name: build-ublue
 on:
-  pull_request:
-    branches:
-      - main
-    paths-ignore:
-      - "**.md"
-      - "**.txt"
+  # Build at 10:20pm UTC every day (1 hr delay after "nvidia" builds).
+  # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule
   schedule:
-    - cron: "20 22 * * *" # 10:20pm everyday (1 hr delay after 'nvidia' builds)
+    - cron: "20 22 * * *"
+  # Build automatically after pushing commits or tags to the "build" branch,
+  # except when the commit only affects "documentation" text files.
+  # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#push
   push:
     branches:
-      - main
+      - build
     paths-ignore:
       - "**.md"
       - "**.txt"
+  # Build when manually triggering this workflow.
+  # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch
   workflow_dispatch:
 
 env:
   IMAGE_REGISTRY: ghcr.io/${{ github.repository_owner }}
 
 jobs:
   push-ghcr:
+    # Only build the branch named "build". Ignore all other branches, to allow
+    # having "development" branches without interfering with GHCR image uploads.
+    if: ${{ github.ref == 'refs/heads/build' }}
     name: Build and push image
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-latest
```

Comment:
These are versioned on purpose so you have to opt in to upgrading your build pipeline. If we deviate here we should document it.

Reply:
@castrojo Yeah but does it make any sense to version the build tools? We're afraid that builds would suddenly stop working if a newer version of the build tools changes in some way, right? But then again, it will also stop building when this image version no longer exists. But I will revert that change. It was the most questionable change in my eyes too.